[REL-1518] Release notes for v24.3.0-alpha.2 (#19003)
cockroachdb/docs: src/current/_includes/releases/v24.3/v24.3.0-alpha.2.md

## v24.3.0-alpha.2

Release Date: October 14, 2024

{% include releases/new-release-downloads-docker-image.md release=include.release %}

<h3 id="v24-3-0-alpha-2-security-updates">Security updates</h3>

> - The parameters for an [HBA config entry]({% link v24.3/security-reference/authentication.md %}#hba-configuration-syntax) for LDAP are now validated when the entry is created or amended, in addition to the validation that happens during an authentication attempt. [#132086][#132086]

Author: @souravcrl PTAL

Reviewer: Can you share if there was a note for LDAP release in 24.2 @mdlinville? We will need to add documentation for LDAP release and the feature wiki also since it was in private preview earlier in 24.2. cc: @pritesh-lahoti 
Additionally the validation during authentication attempt, clients may want to look into setting the cluster setting `server.auth_log.sql_sessions.enabled` to true, which provides exact failure LDAP authentication failure reason.

Author: This is just release notes for the alpha.2. There should be some v24.2 and v24.3 product change issues automatically created in JIRA, which will track the docs work for the LDAP auth changes.

Author: This change has no release note yet in v24.2 -- which v24.2 patch did you expect to see it in? https://www.cockroachlabs.com/docs/releases/v24.2

Reviewer: This is the [backport PR ](https://github.com/cockroachdb/cockroach/pull/127124) which got merged into 24.2.0-rc and I also see the changes [here](https://github.com/cockroachdb/cockroach/blob/blathers/backport-release-24.2.0-rc-128178/pkg/ccl/ldapccl/settings.go). I feel somehow this got missed in the last rlease note then if the release note texts are automatically fetched and we may need to add them. Is there a reason why this was excluded?

Reviewer: Please add ldap authentication to Feature Availability page for v24.2 as that will help provide context for newer feature work/improvements in 24.3 as per our discussion. Thanks for all the help.

> - Added automatic cleanup and validation for dropped roles when [altering default privileges]({% link v24.3/alter-default-privileges.md %}). [#131782][#131782]

Author: @annrpom PTAL

Reviewer 2: @mdlinville 
```suggestion
- Added automatic cleanup and validation for for [default privileges]({link}) that reference dropped roles
```
And then the right link to https://www.cockroachlabs.com/docs/v24.2/security-reference/authorization#default-privileges instead

The current note sounds like the cleanup/validation happens during an `ALTER DEFAULT PRIVILEGE` statement, but it happens during version upgrade/finalization. Otherwise, LGTM!

Author: Thanks, I edited your edit before applying it.

<h3 id="v24-3-0-alpha-2-general-changes">General changes</h3>

- Changed the license `cockroach` is distributed under to the new CockroachDB Software License (CSL). [#131799][#131799] [#131794][#131794] [#131793][#131793]

<h3 id="v24-3-0-alpha-2-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>

> - You can now [authenticate to the DB console API]({% link v24.3/ui-overview.md %}#authentication) by supplying a Java Web Token (JWT) as a Bearer token in the Authorization header. [#130779][#130779]

Author: @pritesh-lahoti PTAL

Reviewer 3: @mdlinville This feature is only required for JPMC right now and will be backported to 23.x.
We need not rush into documenting this feature publicly just yet. We could target it for 25.1.

Author: Hey, this is just for a release note -- you included release note text in your code commits, so it got picked up here by our automation. Are you asking me to remove this note? If so, you probably should not have included release note text in the commit, because each major version that gets the commit will have a draft release note that we will need to remove, and it's very likely that it will be a different writer who is working on that and will have no idea about this history.

<h3 id="v24-3-0-alpha-2-sql-language-changes">SQL language changes</h3>

- To view comments on a type, you can use the new [`SHOW TYPES WITH COMMENT`]({% link v24.3/show-types.md %}#) command. Comments can be added using [`COMMENT ON`]({% link v24.3/comment-on.md %}). [#131183][#131183]
- You can create or alter a [user-defined function (UDF)]({% link v24.3/user-defined-functions.md %}) or [stored procedure (SP)]({% link v24.3/stored-procedures.md %}) with `[EXTERNAL] SECURITY DEFINER` instead of the default `[EXTERNAL] SECURITY INVOKER`. With `SECURITY DEFINER`, the privileges of the owner are checked when the UDF or SP is executed, rather than the privileges of the executor. The `EXTERNAL` keyword is optional and exists for SQL language conformity. [#129720][#129720]

<h3 id="v24-3-0-alpha-2-operational-changes">Operational changes</h3>

> - The following new [metrics]({% link v24.3/metrics.md %}) show details about [replication]({% link v24.2/architecture/replication-layer.md %}) flow control send queue when the [cluster setting]({% link v24.3/cluster-settings.md %}) `kvadmission.flow_control.enabled` is set to `true` and the cluster setting `kvadmission.flow_control.mode` is set to `apply_to_all`.

Author: @kvoli PTAL at this and the next entry

    - `kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue`
    - `kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue`
    - `kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue`
    - `kvflowcontrol.range_controller.count`
    - `kvflowcontrol.send_queue.bytes`
    - `kvflowcontrol.send_queue.count`
    - `kvflowcontrol.send_queue.prevent.count`
    - `kvflowcontrol.send_queue.scheduled.deducted_bytes`
    - `kvflowcontrol.send_queue.scheduled.force_flush`

    [#131857][#131857]

- The following ({% link v24.3/metrics.md %}) have been renamed:
      Previous name                                    | New name-
      -------------------------------------------------|-----------
      `kvflowcontrol.tokens.eval.regular.disconnected` | `kvflowcontrol.tokens.eval.regular.returned.disconnect`
      `kvflowcontrol.tokens.eval.elastic.disconnected` | `kvflowcontrol.tokens.eval.elastic.returned.disconnect`
      `kvflowcontrol.tokens.send.regular.disconnected` | `kvflowcontrol.tokens.send.regular.returned.disconnect`
      `kvflowcontrol.tokens.send.elastic.disconnected` | `kvflowcontrol.tokens.send.elastic.returned.disconnect`

>       [#131857][#131857]

Author: Done! So you can't have the blank line for a sub-llist, but you need it for a table!

<h3 id="v24-3-0-alpha-2-db-console-changes">DB Console changes</h3>

> - The **Table Details** and **Tables** list in the [**Databases page**]({% link v24.3/ui-databases-page.md %}) and the table lists in **Databases** pages now use cache to reuse common data for increased efficiency [#131769][#131769]

Author: @kyle-a-wong PTAL as you reviewed the code change

Reviewer 4: @mdlinville I think we could leave out some implementation specific detail here and leave it a bit more generalized, something like: 
```suggestion
- Revamped the UI and improved performance in the **Databases**, **Tables View**, and **Table Details** pages in the [**Databases page**]({% link v24.3/ui-databases-page.md %})  [#131769][#131769]
```

Author: Thanks, I edited your edit to be even a bit more concise. :)

<h3 id="v24-3-0-alpha-2-bug-fixes">Bug fixes</h3>

- Fixed a bug where JSON values returned by `cockroach` commands using the `--format=sql` flag were not correctly escaped if they contained double quotes within a string. [#131881][#131881]
- Fixed an error that could happen if an [aggregate function]({% link v24.3/functions-and-operators.md %}#aggregate-functions) was used as the value in a [`SET` command]({% link v24.3/set.md %}). [#131891][#131891]
- Fixed a rare bug introduced in v22.2 in which an update of a [primary key]({% link v24.3/primary-key.md %}) column could fail to update the primary index if it is also the only column in a separate column family. [#131869][#131869]

> - Fixed a rare bug where dropping a column of `FLOAT4`, `FLOAT8`, `DECIMAL`, `JSON`, `ARRAY`, or collate `STRING` type stored in a single column family could could prevent subsequent reading of the table if the column family was not the first column family. [#131967][#131967]

Author: ```suggestion
- Fixed a rare bug where dropping a column of `FLOAT4`, `FLOAT8`, `DECIMAL`, `JSON`, `ARRAY`, or collated `STRING` type stored in a single column family could could prevent subsequent reading of the table if the column family was not the first column family. [#131967][#131967]
```
